### PR TITLE
Impove update of the healthcheck in the daemon (scalability optimisation)

### PR DESCRIPTION
**Pull request by ahocquard**
**Description (for Contributor and Core Developer)**

When consuming the queue, a job can run few seconds (less than 5). For example, a calculation fo the completeness when modifying a product model (and impacting children). 

In that case, most of the time was spent at waiting the next update of the healthcheck (due to the sleep in the loop). 

So, the next job to consume was not done before 5 seconds, which is wasted time. 

When there are a lot of small jobs to run, it could lead to scalability problems. For example, a lot of modification of product models. 

This commit improves it by checking more often that the process is still running far more often, but does not update the healthcheck with the same frequency (to not overload the database).

Without this optimization, it generates fails on the CI because:
- some scenarios load fixture data relative to product models (catalog modeling in general)
- it triggers the calculation of the completeness on products/productmodels/etc. So, some jobs are in the queue.
- you run a job in your test, it is not consumed by the daemon, because it is still busy on the jobs triggered when loading fixture data.
- wait time exception 🎉 

Note: the problem could in theory still occur, but the daemon is optimized the best I can without overloading the database with query.

Second commit: add a wait time when catching an exception. It avoids to loop when database is not accessible for example, consuming all the CPU resources.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added Behats                      | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Bundle/BatchQueueBundle/Command/JobQueueConsumerCommand.php [hunk 1]
```diff
@@ -77,13 +83,25 @@ protected function execute(InputInterface $input, OutputInterface $output)
 
                 $process->start();
 
-                do {
-                    $this->getJobExecutionManager()->updateHealthCheck($jobExecutionMessage);
-                    // TODO: standard and error output in a dedicated file for each job execution
+                $this->getJobExecutionManager()->updateHealthCheck($jobExecutionMessage);
+
+                $nbIterationBeforeUpdatingHealthcheck = self::HEALTH_CHECK_INTERVAL * 1000000 / self::RUNNING_PROCESS_CHECK_INTERVAL;
```
Comment (ahocquard) [inline src/Akeneo/Bundle/BatchQueueBundle/Command/JobQueueConsumerCommand.php]: I didn't find better, but it's explicit, any suggestion welcome :)

Comment (Doodoune) [reply src/Akeneo/Bundle/BatchQueueBundle/Command/JobQueueConsumerCommand.php]: typo nbIterationBeforeUpdatingHealth**c**heck - capitalize C

File: src/Akeneo/Bundle/BatchQueueBundle/Command/JobQueueConsumerCommand.php [hunk 2]
```diff
@@ -35,11 +35,17 @@
...
+    /** Interval in seconds before updating health check if job is still running. */
     public const HEALTH_CHECK_INTERVAL = 5;
 
     public const COMMAND_NAME = 'akeneo:batch:job-queue-consumer-daemon';
 
+    /** Interval in microseconds before checking if the process is still running. */
+    private const RUNNING_PROCESS_CHECK_INTERVAL = 200000;
+
+    /** Interval in seconds to wait after an exception occurred.*/
+    private const EXCEPTION_WAIT_INTERVAL = 5;
```
Comment (pierallard) [inline src/Akeneo/Bundle/BatchQueueBundle/Command/JobQueueConsumerCommand.php]: really weird to have Seconds and Microseconds in same level of const... Can you unify plz ? in second, it's more human-readable.

Comment (ahocquard) [reply src/Akeneo/Bundle/BatchQueueBundle/Command/JobQueueConsumerCommand.php]: The goal is to check every X milliseconds with `usleep` so it's necessary and not doable with `sleep`. So it has to be microseconds.
Do you want me to switch the order of the declaration ?

**Pull request merged**